### DISCUSSION TITLE
Libvips php mapim

### DISCUSSION THREAD

**Discussion by cmsaccount** [libvips/libvips#3812] [Help wanted]
Hello,

I need to convert the Imagick code to Libvips. This is a distort transformation horizontally:

```
$image = new Imagick(...);
$points = [ 
0, 0,
0, 0,

0, 10,
10, 10,

100, 0,
100, 0,

100, 10,
110, 10];

$image->distortImage( Imagick::DISTORTION_PERSPECTIVE, $points, TRUE );
```

I know that i must use a method mapim() but i don't understand what arguments to use :(

Thank you!

![1](https://github.com/libvips/libvips/assets/51406194/b9219d6c-005a-4947-8036-8e214f56b36c)

**Comment by jcupitt** [comment]
Hi again,

That's a shear transform, so you can do it with affine.

https://en.wikipedia.org/wiki/Shear_mapping

For example, with this test image:

![PNG_transparency_demonstration](https://github.com/libvips/libvips/assets/580843/ef31fb25-3e04-4471-b580-2cc592296bdd)

I can run:

```
$ vips affine PNG_transparency_demonstration.png x.png "1 0.4 0 1"
```

To make:

![x](https://github.com/libvips/libvips/assets/580843/b3be0def-9060-40b4-8bd9-9f8453269f40)
